document-store/src/main/java/org/hypertrace/core/documentstore/model/subdoc/SubDocumentUpdate.java

```diff
@@ -57,6 +58,10 @@ public static SubDocumentUpdate of(final SubDocument subDocument, final Number v
     return of(subDocument, SubDocumentValue.of(value));
   }
 
+  public static SubDocumentUpdate of(final SubDocument subDocument, final Date date) {
```

Comment:
Can we maintain and manage this as a part of the library implicitly? 
Exposing it outside to the clients could cause incorrect values to be set and may lead to inconsistencies.

Reply:
not sure  how to set the data type for _lastUpdateTime as Date if we don't expose it.